lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -0,0 +1,233 @@
+        """
+        if not self.payload.get("user", None):
+            pytest.skip("skipping accessing user data with invalid token test")
+        # valid token logout
+        response = client.post(
+            f"{server_config.rest_uri}/logout",
+            headers=dict(Authorization="Bearer " + pbench_token),
+        )
+        assert response.status_code == HTTPStatus.OK
+        self.payload["user"] = user
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=self.payload,
+        )
+        assert response.status_code == HTTPStatus.FORBIDDEN
+
+    def test_missing_json_object(self, client, server_config, pbench_token):
+        """
+        Test behavior when no JSON payload is given
+        """
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+        )
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+        assert response.json.get("message") == "Invalid request payload"
+
+    def missing_keys(self, client, server_config, keys, user_ok, pbench_token):
+        """
+        Test behavior when JSON payload does not contain all required keys.
+
+        Note that Pbench will silently ignore any additional keys that are
+        specified but not required.
+       """
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=keys,
+        )
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+        missing = [k for k in self.required_keys if k not in keys]
+        assert (
+            response.json.get("message")
+            == f"Missing required parameters: {','.join(missing)}"
+        )
+
+    def test_bad_dates(self, client, server_config, user_ok, pbench_token):
+        """
+        Test behavior when a bad date string is given
+        """
+        if not self.bad_date_payload:
+            pytest.skip("skipping the bad date test")
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=self.bad_date_payload,
+        )
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+        assert (
+            response.json.get("message")
+            == "Value '2020-19' (str) cannot be parsed as a date/time string"
+        )
+
+    def test_empty_query(
+        self,
+        client,
+        server_config,
+        query_api,
+        user_ok,
+        find_template,
+        build_auth_header,
+    ):
+        """
+        Check proper handling of a query resulting in no Elasticsearch matches.
+        The test will run thrice with different values of the build_auth_header
+        fixture.
```

Comment:
The fact that the test relies on the `build_auth_header` fixture is evident from the code; what needs commenting on is the fact that that fixture is likely to cause the test to be executed repeatedly, because that might be surprising to the reader.  Naming the fixture in the comment should be sufficient to prompt the reader to go look at it if they want the details; so, all that should be need here is to point out that it will trigger multiple runs.